soroban-sdk/src/testutils.rs

```diff
@@ -87,9 +87,28 @@ pub mod budget {
         /// Reset the budget.
         pub fn reset(&mut self) {
             self.0.reset_default();
+        }
+
+        pub fn reset_unlimited(&mut self) {
             self.0.reset_unlimited();
         }
 
+        pub fn reset_limits(&mut self, cpu: u64, mem: u64) {
+            self.0.reset_limits(cpu, mem);
+        }
+
+        pub fn reset_unlimited_cpu(&mut self) {
+            self.0.reset_unlimited_cpu();
+        }
+
+        pub fn reset_unlimited_mem(&mut self) {
+            self.0.reset_unlimited_mem();
+        }
+
+        pub fn reset_inputs(&mut self) {
+            self.0.reset_inputs();
+        }
```

Comment:
I don't know if you'll need the `reset_input` function in any case. Once you call the reset budget it automatically resets inputs.

Reply:
I'd argue that it can be useful in a scenario where you have to call multiple functions in one or more contracts for setup but only want to measure one.  
e.g., in FCA00C, we have the game engine `init` and a user's `solve` but only want to measure the performance for the user's contract. Therefore we use the tests as usual and reset the inputs between the two calls.

Reply:
One could make the point for rust calling `reset_limits` when ever one would call `reset_inputs` but I it feels more clean with an own function imo

Reply:
Sounds good.